fenix/app/src/main/java/org/mozilla/fenix/shopping/ReviewQualityCheckFeature.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping
+
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.fenix.nimbus.FxNimbus
+
+/**
+ * Feature implementation that provides review quality check information for supported product
+ * pages.
+ */
+class ReviewQualityCheckFeature(
+    private val onAvailabilityChange: (isAvailable: Boolean) -> Unit,
+) : LifecycleAwareFeature {
+
+    override fun start() {
+        val isFeatureEnabled = FxNimbus.features.shoppingExperience.value().enabled
+        // Update to use product page detector api in Bug 1840580
```

Comment:
@jonalmeida we talked about removing the extra step of communication of: first fetching "is this a product page", if so also fetch the analysis. what would happen here instead?

(Rahul, Jon might be out for PTO the next couple weeks, not a blocker)

Reply:
Here, is where we would use the `EngineSession.registerDelegate` API and when the page load is complete, we (GV) internally do the "is supported check" and chain that up to fire this delegate when we get a response. That would feed into the callback on the line below.

The WebPushIntegration works the same way (on the engine/runtime level).

Reply:
gotcha, thanks for clarifying!

Reply:
The idea here is to just perform the `isSupportedProductPage` check, which is a local regex check AFAIU, and show the shopping experience icon or not. This would return quite fast. Jared suggested something like this [here](https://phabricator.services.mozilla.com/D180978#6014444).

If you're discussing fetching the product details here as well, I'd like to make a case that we should fetch the product details and recommendations when the user actually opens the bottom sheet, or at least delay it, so we don't fetch super eagerly, as we don't know how many people would eventually open the bottom sheet per "unique supported product page load".

I'm curious what are the advantages of removing the extra step of communication of `isSupportedProductPage`?

Reply:
> I'm curious what are the advantages of removing the extra step of communication of `isSupportedProductPage`?

@rahulsainani I don't see any response here, so just to close the loop: we assumed that we wanted to request product information regardless of the embedder checking. Having seen your demo in Slack recently, it looks like having the separate `isSupportedProductPage` makes sense for an indicator. A product check after that then makes sense. 🙂